### DISCUSSION TITLE
Generate KafkaUser secret

### DISCUSSION THREAD

**Discussion by XBeg9** [strimzi/strimzi-kafka-operator#4861] [Q&A] [answered]
Is there a way to generate or set a particular password for **KafkaUser**?
I was able to mimic the KafkaUser secret, but keep getting:
`org.apache.kafka.common.errors.SaslAuthenticationException: Authentication failed during authentication due to invalid credentials with SASL mechanism SCRAM-SHA-512`

Spent an enormous time trying to find any particular documentation about this, but no lack. 

Thanks

**Comment by scholzj** [comment, marked as answer]
Normally, you should be able to create the secret with the right structure first before creating the `KafkaUser` resource. When the operator sees it then, it will just reuse it and set the password from it. It is not documented - it is really just a hack. One think to double check is that you got all the base64 encodings right. That seems to be common source of issues when working with secrets.

**Reply by XBeg9** [reply]
I did exactly what you've described, probably could be the race-condition while creating KafkaUser resource and secret at the same time, let me delay it and see how it goes. Thanks again @scholzj

**Reply by scholzj** [reply]
Give it another try and if it still doesn't work let me know, I can have a look if something changed. But nobody else reported any issues with this.

**Reply by XBeg9** [reply]
I can approve, It does work if you wait until the secret is created and only then create KafkaUser resource. Thanks @scholzj
